applications/StructuralMechanicsApplication/custom_strategies/custom_builder_and_solver/explicit_builder_and_solver.hpp

```diff
@@ -217,32 +191,22 @@ class ExplicitBuilderAndSolver
         ProcessInfo& rCurrentProcessInfo      = rModelPart.GetProcessInfo();
         ConditionsArrayType& pConditions      = rModelPart.Conditions();
 
-        #ifdef _OPENMP
-            int number_of_threads = omp_get_max_threads();
-        #else
-            int number_of_threads = 1;
-        #endif
-
-        vector<unsigned int> condition_partition;
-        OpenMPUtils::CreatePartition(number_of_threads, pConditions.size(), condition_partition);
+        typename ConditionsArrayType::ptr_iterator it_begin=pConditions.ptr_begin();
+        #pragma omp parallel for
+        for (size_t i =0;i<pConditions.size();++i)
+        {
+            LocalSystemVectorType RHS_Condition_Contribution = LocalSystemVectorType(0);
+            Element::EquationIdVectorType equation_id_vector_dummy; //Dummy
 
+            pScheme->Condition_Calculate_RHS_Contribution(*it_begin, RHS_Condition_Contribution,
+             equation_id_vector_dummy, rCurrentProcessInfo);
+            it_begin++;
```

Comment:
take care, this is WRONG!! it_begin is a shared variable and you are actualizing it!!

Reply:
can I do the same as it is done in scheme.h ?

#pragma omp parallel for
for (int i=0;i<r_elements.size();++i)  
        {
            auto it_elem = rModelPart.ElementsBegin() + i;
            //Getting nodal mass and inertia from element
            Vector dummy_vector;
            // this function needs to be implemented in the respective 
            // element to provide inertias and nodal masses 
            it_elem->AddExplicitContribution(dummy_vector,RESIDUAL_VECTOR,NODAL_INERTIA,r_current_process_info);
        }

Reply:
as an example